go/vt/topo/shard_test.go

```diff
@@ -329,29 +328,16 @@ func TestUpdateServedTypesMap(t *testing.T) {
 		t.Fatalf("migrate master failed: %v", err)
 	}
 
-	// try to migrate master away, see it fail
-	if err := si.UpdateServedTypesMap(topodatapb.TabletType_MASTER, nil, true); err == nil || err.Error() != "cannot migrate MASTER away from ks/sh until everything else is migrated. Make sure that the following types are migrated first: RDONLY, REPLICA" {
```

Comment:
It looks like we don't have a test case for this anymore

Reply:
It's because `UpdateServedTypesMap` doesn't do this check any more. The check has moved to `masterMigrateServedTypes`. And test for it is in resharding.py: https://github.com/vitessio/vitess/blob/master/test/resharding.py#L935.

Reply:
Ah perfect.